spec.bs

```diff
@@ -0,0 +1,369 @@
+
+  <h4 id="topics-allowable-fetch-request-header">Topics-allowable fetch() request</h4>
+  The following algorithm determines whether topics are allowed to be sent on a fetch() request header:
+    1. If the request's RequestInit parameter does not contain |{browsingTopics: true}|:
+        1. Return false.
+    1. Determine the [=topics-caller-context=] |C|.
+    1. If |C|.caller-origin is not <a href="https://www.w3.org/TR/secure-contexts/#potentially-trustworthy-origin">potentially trustworthy</a>:
+        1. Return false.
+    1. If the fetch() is not initiated from a Window context:
+        1. Return false.
+    1. If the fetch() initiator document is not in [=secure context=]:
+        1. Return false.
+    1. For each feature |F| in {"browsing-topic", "interest-cohort"}:
+        1. Run the <a href="https://www.w3.org/TR/permissions-policy-1/#define-inherited-policy">inherited policy for feature in browsing context</a> algorithm with input arguments |feature|: |F|, |origin|: the fetch() URL's origin, <var ignore=''>browsingContext</var>: the fetch initiator document. If the algorithm returns "Disabled":
+            1. Return fasle.
+    1. If the user preference setting disallows the access to topics from |C|.caller-origin and |C|.page.top-level-origin:
+        1. Return false.
+    1. Return true.
+
+  Note: In Chrome's experimentation phase, fetch() request will additioanlly require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in the initiator Document for the request to be topics-allowable.
+
+  <h3 id="derive-topics-header-value-header">Derive Topics Header Value</h3>
+  The <dfn>topics header value</dfn> is the Structured Fields representation of the list of [=topic ids=] along with versions as Parameters.
+
+  <div class="example">
+  This is an example value that consists two topics, 1 and 7, along with their version information:
+
+  <code>1;version="chrome.1:1:2";config_version="chrome.1";model_version="2";taxonomy_version="1", 7;version="chrome.1:1:2";config_version="chrome.1";model_version="2";taxonomy_version="1"</code>
+  </div>
+
+  <h3 id="handle-http-request-response-header-header">Handle HTTP Request/Response Header</h3>
+  For each <a href="#topics-allowable-http-request">topics allowable</a> HTTP request:
+  1. Determine the [=topics-caller-context=] |C|.
+  1. Let |result| be the result from running the [=topics for caller=] algorithm, with |C| as input.
+  1. If |result| is not empty:
+    1. Derive [=topics header value=] |V| from |result|, and append "Sec-Browsing-Topics: |V|" to the request header.
+
+  For the corresponding response:
```

Comment:
This algorithm needs to be written to use the proper mechanisms for retrieving and parsing headers. You can see an example at https://html.spec.whatwg.org/#navigating-across-documents:origin-agent-cluster . You will need to state more precisely where this algorithm runs, instead of just "for the corresponding response"; does it run on all responses, and thus should patch a certain section of https://fetch.spec.whatwg.org/ ? Which section?